pyproject.toml

```diff
@@ -10,129 +25,202 @@ authors = [{ name = "Catalyst Cooperative", email = "pudl@catalyst.coop" }]
-    "grpcio-status==1.73.1", # Required by dagster, binary dependencies are flaky
-    "hypothesis>=6.123",
-    "jellyfish>=1",
-    "jinja2>=3.1",
-    "jupyter",
-    "matplotlib>=3.10",
-    "mlflow>=3.6",
-    "nbconvert>=7",
-    "nbformat>=5.10",
-    "networkx>=3.6",
-    "numba>=0.63.1", # pandas[performance]
-    "numexpr>=2.10", # pandas[performance]
-    "numpy>=2.3",
-    "packaging>=25",
-    "pandas>=2.3",
-    "pandera>=0.27",
-    "polars>=1.36",
-    "pre-commit>=4",
-    "pyarrow>=20,<21", # pandas[parquet]; <21 for Kaggle geopandas compatibility
-    "pydantic>=2.12",
-    "pydantic-settings>=2.11",
-    "pytest>=9",
-    "pytest-console-scripts>=1.4",
-    "pytest-cov>=7",
-    "pytest-mock>=3.15",
-    "pytest-order>=1.3",
-    "pytest-xdist>=3.8",
-    "python-calamine>=0.6.1", # pandas[excel]
-    "python-dotenv>=1",
-    "pytz>=2025",
-    "pyyaml>=6",
-    "readthedocs-sphinx-ext>=2",
-    "requests>=2.32",
-    "responses>=0.25",
-    "ruff>=0.14",
-    "s3fs>=2025",
-    "scikit-learn>=1.8",
-    "scipy>=1.16",
-    "Shapely>=2.1.1",
-    "splink>=4.0.10",
-    "sphinx>=8",
-    "sphinx-autoapi>=3.6.1",
-    "sphinx-issues>=5",
-    "sphinx-reredirects>=1",
-    "sphinx-tabs>=3.4",
-    "sphinxcontrib_bibtex>=2.6",
-    "sphinxcontrib_googleanalytics>=0.4",
-    "sqlalchemy>=2",
-    "timezonefinder>=8",
-    "typos>=1.40", # Spell-checker
-    "universal_pathlib>=0.3.7",
-    "urllib3>=2.4",
-    "xlsxwriter>=3.2", # pandas[excel]
-]
-classifiers = [
-    "Development Status :: 5 - Production/Stable",
-    "Environment :: Console",
-    "Intended Audience :: Science/Research",
-    "License :: OSI Approved :: MIT License",
-    "Natural Language :: English",
-    "Operating System :: OS Independent",
-    "Programming Language :: Python :: 3 :: Only",
-    "Programming Language :: Python :: 3.13",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python",
-    "Topic :: Scientific/Engineering",
-]
-keywords = [
-    "analysis",
-    "climate change",
-    "coal",
-    "data",
-    "economics",
-    "eia 860",
-    "eia 861",
-    "eia 923",
-    "electricity",
-    "emissions",
-    "energy",
-    "epa ampd",
-    "epa cems",
-    "ferc form 1",
-    "ferc form 2",
-    "ferc form 6",
-    "ferc form 60",
-    "ferc form 714",
-    "finance",
-    "fossil fuel",
-    "natural gas",
-    "oil",
-    "policy",
-    "utility",
-]
+# Dependencies are managed via pixi and installed from conda-forge.
+# See [tool.pixi.dependencies] and [tool.pixi.feature.*.dependencies] below.
+# This package is an application, not a library, and should only be installed
+# from the repository in editable mode using: pixi install
+# All optional dependencies are now managed via pixi features.
+# See [tool.pixi.feature.*.dependencies] below.
+dependencies = []
+
+[tool.pixi.workspace]
+name = "catalystcoop.pudl"
+platforms = ["linux-64", "osx-64", "osx-arm64"]
+channels = ["conda-forge"]
+
+[tool.pixi.dependencies]
```

Comment:
For now I've consolidated all of the dependencies into a single large environment. It solves fast and keeps things simple.  We can break it up into smaller "features" with optional dependencies / environments later if we want to.